src/lib/containers/AccessApprovalsTable.tsx

```diff
@@ -0,0 +1,142 @@
+import moment from 'moment'
+import React, { useState } from 'react'
+import { Button, Table } from 'react-bootstrap'
+import { SMALL_USER_CARD } from '../utils/SynapseConstants'
+import {
+  AccessApprovalSearchRequest,
+  AccessApprovalSearchSort,
+  AccessApprovalSortField,
+  Direction,
+  SortDirection,
+} from '../utils/synapseTypes'
+import Typography from '../utils/typography/Typography'
+import UserCard from './UserCard'
+import { PRODUCTION_ENDPOINT_CONFIG } from '../utils/functions/getEndpoint'
+import { useSearchAccessApprovalsInfinite } from '../utils/hooks/SynapseAPI/dataaccess/useAccessApprovals'
+import { SynapseSpinner } from './LoadingScreen'
+import SortIcon from '../assets/icons/Sort'
+import { formatDate } from '../utils/functions/DateFormatter'
+
+export type AccessApprovalsTableProps = {
+  accessorId: string
+  accessRequirementId?: string
+}
+export const AccessApprovalsTable: React.FunctionComponent<
+  AccessApprovalsTableProps
+> = ({ accessorId, accessRequirementId }) => {
+  const [isDescending, setIsDescending] = useState(false)
+  const [sort, setSort] = useState<AccessApprovalSearchSort[]>()
+
+  const searchRequest: AccessApprovalSearchRequest = {
+    accessorId,
+    accessRequirementId,
+    sort,
+  }
+
+  const { data, hasNextPage, fetchNextPage, isFetching } =
+    useSearchAccessApprovalsInfinite(searchRequest)
+  const accessApprovals = data?.pages.flatMap(page => page.results) ?? []
+  const onSort = (field: AccessApprovalSortField, isDescending: boolean) => {
+    const direction: SortDirection = isDescending
+      ? ('DESC' as SortDirection)
+      : ('ASC' as SortDirection)
+    setSort([{ field, direction }])
+    setIsDescending(!isDescending)
+  }
```

Comment:
Update this to reflect that we got rid of `isDescending`.

The behavior here subtly changes too; if we switch the sorted field, it will always start 'DESC', rather than always switching directions even when you change fields. This is more predictable from a user's perspective

```suggestion
  const onSort = (field: AccessApprovalSortField) => {
    if (sort.field === field) {
      setSort({ field, direction: sort.direction === SortDirection.DESC ? SortDirection.ASC : SortDirection.DESC })
    } else {
      setSort({ field, direction: SortDirection.DESC })
    }
  }
```